### PR TITLE
Remove legacy parse

### PR DESCRIPTION
**Pull request by tonyyang-svail**

### CODE REVIEW

File: pkg/database/db.go [hunk 1]
```diff
@@ -33,6 +33,11 @@ type DB struct {
 	*sql.DB
 }
 
+// ConnectionString returns the connection string in driver_type://data_source_name format
```
Comment (wangkuiyi) [inline pkg/database/db.go]: This function already exists, named URL()

File: pkg/verifier/verifier_test.go [hunk 2]
```diff
@@ -22,40 +23,53 @@ import (
 	"sqlflow.org/sqlflow/pkg/parser"
 )
 
+func parseOneStatement(statement string) (*parser.SQLFlowSelectStmt, error) {
```
Comment (wangkuiyi) [inline pkg/verifier/verifier_test.go]: Duplicated definition as in ir_generator_test.go

File: pkg/sql/ir_generator_test.go [hunk 3]
```diff
@@ -25,11 +26,21 @@ import (
 	"sqlflow.org/sqlflow/pkg/sql/ir"
 )
 
+func parseOneStatement(statement string) (*parser.SQLFlowSelectStmt, error) {
```
Comment (wangkuiyi) [inline pkg/sql/ir_generator_test.go]: This function duplicates with function `ParseOneStatement` in `parser.go`.

**Review by wangkuiyi — COMMENTED** [review]
Thanks to verifying that we could remove LegacyParse and replace it by parseOneStatement.

Two suggestions:

- `ParseOneStatement` is already there in `parser.go`; there is no need to add them, duplicated in `ir_generator_test.go` and `verifier_test.go`.
- `ParseOneStatement` doesn't parse a statement, but parse a program with one and only one statement. Given that a set with only one element is known as a **unit set**. How about we rename it into `ParseUnitProgram`?

File: pkg/sql/ir_generator_test.go (hunk 3 above)
Comment (tonyyang-svail) [reply pkg/sql/ir_generator_test.go]: Thanks for reminding me.

File: pkg/database/db.go (hunk 1 above)
Comment (tonyyang-svail) [reply pkg/database/db.go]: Thanks for pointing it out!

**Comment by tonyyang-svail** [comment]
> ParseOneStatement doesn't parse a statement, but parse a program with one and only one statement. Given that a set with only one element is known as a unit set. How about we rename it into ParseUnitProgram?

Sounds good. I will do that in the next PR.

File: pkg/verifier/verifier_test.go (hunk 2 above)
Comment (tonyyang-svail) [reply pkg/verifier/verifier_test.go]: Removed.

**Review by typhoonzero — APPROVED** [review]

**Pull request merged**